### ISSUE TITLE
Syntax error in rails when parsing secrets.yml

### ISSUE THREAD

**Issue by dgmora** [Shopify/bootsnap#71]
Hi guys, 

I wanted to give bootsnap a try, but when I wanted to run the tests, I've got the following error in [here](https://github.com/rails/rails/blob/master/railties/lib/rails/secrets.rb#L42):
```
/usr/local/rvm/gems/ruby-2.3.3@global/gems/railties-5.1.2/lib/rails/secrets.rb:45: invalid multibyte char (US-ASCII)
/usr/local/rvm/gems/ruby-2.3.3@global/gems/railties-5.1.2/lib/rails/secrets.rb:42: syntax error, unexpected end-of-input, expecting tSTRING_CONTENT or tSTRING_DBEG or tSTRING_DVAR or tSTRING_END
        <<-end_of_template.strip_heredoc
                          ^
```

The tests run inside docker. `secrets.yml` is correct. There is string interpolation inside the yml  `<%= ENV["SOMETHING"] %>`) in case it matters

**Comment by GermanDZ** [comment]
same here… Rails 5.2, Ruby 2.3

**Comment by sungsong88** [comment]
I'd like to try bootsnap as well, but my rails throws this:
```
| /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `block in load_missing_constant': uninitialized constant Rails::Secrets (NameError)
| 	from /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:11:in `block in with_bootsnap_fallback'
| 	from /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:17:in `without_bootsnap_cache'
| 	from /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:11:in `rescue in with_bootsnap_fallback'
| 	from /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:6:in `with_bootsnap_fallback'
| 	from /usr/local/bundle/gems/bootsnap-1.1.2/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `load_missing_constant'
| 	from /usr/local/bundle/gems/railties-5.1.2/lib/rails/application.rb:391:in `secrets'
```

**Comment by dgmora** [comment]
So the base image I was using is `passenger-ruby`. I had not set any default locale specifically, so `US-ASCII` is used. The rails app is configured to use utf-8 though. I stopped seeing this error after I added 
```
RUN locale-gen en_US.UTF-8
ENV LANG en_US.UTF-8
ENV LANGUAGE en_US:en
ENV LC_ALL en_US.UTF-8
```
in the dockerfile.

This is fixable on my side, but it could be considered a bug as well, as the app behaves differently with or without bootsnap.
